main/hgrep.hs

```diff
@@ -25,12 +28,30 @@ main = do
       IO.hPutStrLn IO.stderr er
       exitWith (ExitFailure 2)
     Right q -> do
+      -- if no directory provided -> search in current directory
+      files <- case cmdFiles opts of
+                 [] -> pure <$> D.getCurrentDirectory
+                 x  -> pure x
+      allFiles <- foldMap getAllHsFiles files
       found <-
         fmap sum $
-          for (cmdFiles opts) $ \fp ->
+          for allFiles $ \fp -> do
             hgrep (HGrep.PrintOpts colour (cmdLineNums opts)) q fp
       exitWith (exitCode found)
 
+getAllHsFiles :: FilePath -> IO [FilePath]
+getAllHsFiles fp = do
+  isDir <- D.doesDirectoryExist fp
+  if isDir
+  then do
+       fs <- L.map (fp FP.</>) <$> D.listDirectory fp
```

Comment:
Same comment as [above](https://github.com/thumphries/hgrep/pull/43/files#r144189266).

Reply:
Agree with this.

Reply:
Much better, thanks :smiley: